File: LeetCode_Python/pow(x, n).py
class Solution(object):
    def myPow(self, x, n):
        if n == 0 : return 1

        if n < 0:
            return 1 / self.myPow(x, -n)
        elif n % 2 == 0:
            return self.myPow(x * x, n // 2)
        else:
            return x * self.myPow(x * x, n // 2)

x = 2.10000
n = 3
sol = Solution()
result = sol.myPow(x, n)
print(f"{result:.5f}")

File: LeetCode_C/pow(x, n).c
#include <stdio.h>

double myPow(double x, int n) {
    if (n == 0) return 1;

    if (n < 0) {
        return 1 / myPow(x, -n);
    } else if (n % 2 == 0) {
        return myPow(x * x , n / 2);
    } else {
        return x * myPow(x * x, n / 2);
    }
}

int main() {
    double x = 2.00000;
    int n = 10;
    double result = myPow(x, n);
    printf("%.5f", result);
    return 0;
}